var/spack/repos/builtin/packages/zlib-ng/package.py

```diff
@@ -76,8 +76,6 @@ def libs(self):
     def flag_handler(self, name, flags):
         if name == "cflags" and self.spec.satisfies("+pic build_system=autotools"):
             flags.append(self.compiler.cc_pic_flag)
-        if name == "ldflags" and self.spec.satisfies("%cce@17"):
-            flags.append("-Wl,--undefined-version")
```

Comment:
If the build didn't fail after failure to create a shared library, then that's a build system problem. Report that upstream instead.

Reply:
> not clear what you mean by "hide"

It is a legitimate problem for certain symbol versions to be referenced that do not exist. It could prevent LD_PRELOAD=/some/other/libz.so to work. You're disabling a strict check for that, so future regressions will not be caught.

Reply:
> If the build didn't fail after failure to create a shared library, then that's a build system problem. 

I think it's our problem if there isn't an upstream fix (we often apply patches within the Spack repo that should be upstreamed, and we remove them when the upstream fix lands): we should avoid the broken versions (and make sure we can pick a combination that works for `zlib-ng+shared@2.1.6:%cce`), or we should detect failed installs, or we should keep this change.

> Report that upstream instead.

I don't think that has to be mutually exclusive with addressing things here in the meantime.

> It is a legitimate problem for certain symbol versions to be referenced that do not exist.

In other words:

1. If we keep everything as-is, we have a potentially broken `zlib-ng+shared` in some cases, but it exists (and seems usable), furthermore the constraint exists as an indicator of something wrong
2. If we apply everything in this PR as-is, then we have a broken `zlib-ng+shared`, and there is no indicator within the `zlib-ng` package that this might be the case

I think [1] is preferable for no other reason than there is at least some indication of something strange going on with `zlib-ng`. I think I've also suggested a couple reasonable alternatives to [1] and [2]

Reply:
> in some cases

you don't know which case